lib/volcanic/authenticator/base.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
```

Comment:
this is a bit of an anti-pattern for a couple of reasons:
- this file now needs to know about all of your code
- this encourages "surprise dependencies" between classes

Consider breaking it up in to separate sub-modules that just deal with that area of responsibility.

For example, a file called `warden.rb` that looked something like:
```
require_relative 'warden/allow_always'
require_relative 'warden/validate_token_always'
require_relative 'warden/validate_token_present'

module Volcanic::Warden; end
```
then, from here, you might do `require_relative 'warden'`

To help with unpicking the "surprise dependencies", classes should require what they need. By doing all of the requires here, you make things dependant on the load order in this class, rather than the class that actually needs it. This helps us to avoid issues like somebody changing the ordering in this file and accidentally breaking things.

For example, `v1/principal` only works if you have already loaded `v1/resource`. This means that `v1/principal` should have a `require_relative 'v1/resource'`. Once all that is done, you would only need to require `v1/principal` from here, and not all of its dependencies.

Reply:
alright noted about this. few changes made to achieve this. now we have `v1`, `config` and `warden` submodule, which contains is own classes.